NGitLab/Impl/MilestoneClient.cs

```diff
@@ -12,10 +12,16 @@ public class MilestoneClient : IMilestoneClient
         private readonly API _api;
         private readonly string _milestonePath;
 
+        [Obsolete("Use long or namespaced path string as id instead.")]
         internal MilestoneClient(API api, MilestoneScope scope, int id)
+            : this(api, scope, (long)id)
+        {
+        }
+
+        internal MilestoneClient(API api, MilestoneScope scope, IdOrNamespacedPath projectId)
         {
             _api = api;
-            _milestonePath = $"/{scope.ToString().ToLowerInvariant()}/{id.ToStringInvariant()}/milestones";
+            _milestonePath = $"/{scope.ToString().ToLowerInvariant()}/{projectId.ValueAsUriParameter}/milestones";
             Scope = scope;
         }
```

Comment:
@Toa741 A base type of `ProjectId` and `GroupId` will make some methods reusable.